cli/src/merge_tools/internal.rs

```diff
@@ -0,0 +1,610 @@
+                lines: make_section_changed_lines(&contents, scm_record::ChangeType::Added),
+            }),
+
+            (FileContents::Absent, FileContents::Binary { hash, num_bytes }) => {
+                sections.push(scm_record::Section::Binary {
+                    is_checked: false,
+                    old_description: None,
+                    new_description: Some(Cow::Owned(describe_binary(hash.as_deref(), num_bytes))),
+                })
+            }
+
+            (
+                FileContents::Text {
+                    contents,
+                    hash: _,
+                    num_bytes: _,
+                },
+                FileContents::Absent,
+            ) => sections.push(scm_record::Section::Changed {
+                lines: make_section_changed_lines(&contents, scm_record::ChangeType::Removed),
+            }),
+
+            (
+                FileContents::Text {
+                    contents: old_contents,
+                    hash: _,
+                    num_bytes: _,
+                },
+                FileContents::Text {
+                    contents: new_contents,
+                    hash: _,
+                    num_bytes: _,
+                },
+            ) => {
+                sections.extend(make_diff_sections(&old_contents, &new_contents)?);
+            }
+
+            (
+                FileContents::Text {
+                    contents: _,
+                    hash: old_hash,
+                    num_bytes: old_num_bytes,
+                }
+                | FileContents::Binary {
+                    hash: old_hash,
+                    num_bytes: old_num_bytes,
+                },
+                FileContents::Text {
+                    contents: _,
+                    hash: new_hash,
+                    num_bytes: new_num_bytes,
+                }
+                | FileContents::Binary {
+                    hash: new_hash,
+                    num_bytes: new_num_bytes,
+                },
+            ) => sections.push(scm_record::Section::Binary {
+                is_checked: false,
+                old_description: Some(Cow::Owned(describe_binary(
+                    old_hash.as_deref(),
+                    old_num_bytes,
+                ))),
+                new_description: Some(Cow::Owned(describe_binary(
+                    new_hash.as_deref(),
+                    new_num_bytes,
+                ))),
+            }),
+
+            (FileContents::Binary { hash, num_bytes }, FileContents::Absent) => {
+                sections.push(scm_record::Section::Binary {
+                    is_checked: false,
+                    old_description: Some(Cow::Owned(describe_binary(hash.as_deref(), num_bytes))),
+                    new_description: None,
+                })
+            }
+        }
+
+        files.push(scm_record::File {
+            old_path: None,
+            path: Cow::Owned(changed_path.to_fs_path(Path::new(""))),
```

Comment:
Does scm-record treat this as a path in some way? For example, does it try to read from it? If not, should it be a `String` instead of a `Cow<PathBuf>` (I'm guessing that's what it is)? (Obviously out of scope for this PR.)